homeassistant/components/aemet/weather_update_coordinator.py

```diff
@@ -112,130 +112,33 @@ def format_int(value) -> int | None:
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        aemet: AEMET,
+    ) -> None:
         """Initialize coordinator."""
-        super().__init__(
-            hass, _LOGGER, name=DOMAIN, update_interval=WEATHER_UPDATE_INTERVAL
-        )
-
-        self._aemet = aemet
-        self._station = None
-        self._town = None
-        self._latitude = latitude
-        self._longitude = longitude
-        self._station_updates = station_updates
-        self._data = {
-            "daily": None,
-            "hourly": None,
-            "station": None,
-        }
+        self.aemet = aemet
 
-    async def _async_update_data(self):
-        data = {}
-        async with timeout(120):
-            weather_response = await self._get_aemet_weather()
-        data = self._convert_weather_response(weather_response)
-        return data
-
-    async def _get_aemet_weather(self):
-        """Poll weather data from AEMET OpenData."""
-        weather = await self._get_weather_and_forecast()
-        return weather
-
-    async def _get_weather_station(self):
-        if not self._station:
-            self._station = (
-                await self._aemet.get_conventional_observation_station_by_coordinates(
-                    self._latitude, self._longitude
-                )
-            )
-            if self._station:
-                _LOGGER.debug(
-                    "station found for coordinates [%s, %s]: %s",
-                    self._latitude,
-                    self._longitude,
-                    self._station,
-                )
-        if not self._station:
-            _LOGGER.debug(
-                "station not found for coordinates [%s, %s]",
-                self._latitude,
-                self._longitude,
-            )
-        return self._station
-
-    async def _get_weather_town(self):
-        if not self._town:
-            self._town = await self._aemet.get_town_by_coordinates(
-                self._latitude, self._longitude
-            )
-            if self._town:
-                _LOGGER.debug(
-                    "Town found for coordinates [%s, %s]: %s",
-                    self._latitude,
-                    self._longitude,
-                    self._town,
-                )
-        if not self._town:
-            _LOGGER.error(
-                "Town not found for coordinates [%s, %s]",
-                self._latitude,
-                self._longitude,
-            )
-            raise TownNotFound
-        return self._town
-
-    async def _get_weather_and_forecast(self):
-        """Get weather and forecast data from AEMET OpenData."""
-
-        await self._get_weather_town()
-
-        daily = await self._aemet.get_specific_forecast_town_daily(
-            self._town[AEMET_ATTR_ID]
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=WEATHER_UPDATE_INTERVAL,
```

Comment:
Is the data that is returned comparable with `__eq__` and may be the same ?

If so you can avoid state writes by setting `always_update=False`

Reply:
> Is the data that is returned comparable with `__eq__` and may be the same ?
> 
> If so you can avoid state writes by setting `always_update=False`

No, it's not comparable using `__eq__`.
However, it will probably be comparable after the full coordinator refactor @ https://github.com/home-assistant/core/pull/99026.